bigml/modelfields.py

```diff
@@ -138,6 +139,77 @@ def check_model_fields(model):
     return False
 
 
+def get_datetime_subfields(fields):
+    """ From a dictionary of fields, returns another dictionary
+    with the subfields from each datetime field
+
+    """
+    subfields = {}
+    for fid, finfo in fields.items():
+        if (finfo.get('parent_optype', False) == 'datetime'):
+            parent_name = ".".join(finfo["name"].split(".")[:-1])
+            subfield = {fid: finfo["name"].split(".")[-1]}
+            if parent_name in subfields.keys():
+                subfields[parent_name].update(subfield)
+            else:
+                subfields[parent_name] = subfield
+    return subfields
+
+
+def expand_date(date, subfields, timeformats):
+    """ Retrieves all the values of the subfields from
+    a given date
+
+    """
+    expanded = {}
+    try:
+        parsed_date = chronos.parse(date, format_names=timeformats)
+    except ValueError:
+        return {}
+    for fid, ftype in subfields.items():
```

Comment:
this can be improved by mapping the datatype to the name of the attribute in the cases where they are not identical